hadoop-crypto/src/test/java/com/palantir/hadoop/PathConvertingFileSystemTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.hadoop;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.base.Function;
+import java.net.URI;
+import java.util.Arrays;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.junit.Before;
+import org.junit.Test;
+
+public final class PathConvertingFileSystemTest {
+
+    private static final Path PATH = new Path("/dummy/path");
+    private static final Path DELEGATE_PATH = new Path("/delagate/path");
+    private static final Path RETURN_PATH = new Path("/return/path");
+    private static final URI DUMMY_URI = URI.create("/dummy/uri");
+
+    private FileSystem delegate;
+    private FSDataInputStream inputStream;
+    private FSDataOutputStream outputStream;
+    private PathConvertingFileSystem convertingFs;
+
+    @Before
+    public void before() {
+        delegate = mock(FileSystem.class);
+        inputStream = mock(FSDataInputStream.class);
+        outputStream = mock(FSDataOutputStream.class);
+        convertingFs = new PathConvertingFileSystem(delegate, DelegatePathFunc.INSTANCE, ReturnPathFunc.INSTANCE);
+    }
+
+    @Test
+    public void getUri() throws Exception {
+        when(delegate.getUri()).thenReturn(DUMMY_URI);
+        URI actualUri = convertingFs.getUri();
+
+        assertThat(actualUri, is(DUMMY_URI));
+    }
+
+    @Test
+    public void open() throws Exception {
+        when(delegate.open(DELEGATE_PATH, 0)).thenReturn(inputStream);
+        FSDataInputStream actualStream = convertingFs.open(PATH, 0);
+
+        assertThat(actualStream, is(inputStream));
```

Comment:
started that way but realized I also want to verify that the return value is correct since it's possible to call the delegate correctly without propagating what the delegate returns.